### PR TITLE
Refactor auth

### PR DESCRIPTION
**Pull request by connieyu18**
- Removed tokens in localStorage, and moved to storing tokens in httpOnly cookies to improve security

### CODE REVIEW

File: client/src/App.js
Comment (Tahoum) [inline client/src/App.js:15]: You need to add `delete` to be able to log out!

File: server/app.js
Comment (Tahoum) [inline server/app.js]: You can remove the commented code if you don't need it

File: server/routes/auth/index.js
Comment (Tahoum) [inline server/routes/auth/index.js]: You can add the `expires` option, also create `cookieOptions` object and reuse it.

File: server/routes/auth/index.js
Comment (Tahoum) [inline server/routes/auth/index.js]: You don't need to return the token in the response since you have it set in the cookie.

**Review by Tahoum — COMMENTED** [review]
Good job for choosing to implement `CSRF` tokens.
I noticed that you don't clear the cookie on log out! Also, address the comments and once you're done let us know in the channel so we can give you a second review 🙂

File: client/src/App.js
Comment (connieyu18) [reply client/src/App.js:15]: Thanks for catching that! I added delete to logout in thunk creators

File: server/routes/auth/index.js
Comment (connieyu18) [reply server/routes/auth/index.js]: Thanks! I will remember that next time

File: server/routes/auth/index.js
Comment (Mayo19) [inline server/routes/auth/index.js:85]: The [documentation for `res.clearCookie`](http://expressjs.com/en/api.html#res.clearCookie) says that compliant browsers/clients will only clear the cookie if the provided `options` are identical to those given when setting the cookie with `res.cookie()` (aside from `expires` and `maxAge`).

**Review by Mayo19 — APPROVED** [review]
Looks great! Just left a small comment.

Approved so feel free to merge once you've addressed the comment.

File: server/routes/auth/index.js
Comment (connieyu18) [reply server/routes/auth/index.js:85]: Since I only added httpOnly as option besides maxAge, can I change it to " res.clearCookie("token", { httpOnly: true })"?

**Pull request merged**
